static-assets/components/cstudio-forms/controls/date-time.js

```diff
@@ -28,6 +28,8 @@ CStudioForms.Controls.DateTime =
     this.value = '_not-set';
     this.form = form;
     this.id = id;
+    this.locale = CrafterCMSNext.system.store.getState().uiConfig.locale;
+    this.hour12 = this.locale.dateTimeFormatOptions?.hour12 ?? true;
```

Comment:
Asynchrony may be an issue here. This would likely pick up the default and not realised when the actual config has loaded.